### PR TITLE
created 2 more actions using chat GPT

### PR DESCRIPTION
**Pull request by dougsillars**
## Description
Please include a summary of the change, motivation and context.
2 new actions:

1. add a tag to a EC2 instance.   Supply region, instanceID, tag key and tag value to tag an instance.

2. GCP: list all VMs in a region and project.  The output also tells you if the VMs are publicly available or not.

### CODE REVIEW

File: GCP/legos/gcp_list_vms_access/gcp_list_vms_access.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "action_title": "List all GCP VMs and if Publicly Accessible",
+    "action_description": "List all GCP buckets",
+    "action_type": "LEGO_TYPE_GCP",
+    "action_entry_function": "gcp_list_VMs",
+    "action_needs_credential": true,
+    "action_supports_poll": true,
+    "action_supports_iteration": true,
```
Comment (shloka-bhalgat-unskript) [inline GCP/legos/gcp_list_vms_access/gcp_list_vms_access.json:8]: This file is missing - `action_output_type": "ACTION_OUTPUT_TYPE_?`

File: AWS/legos/aws_tag_ec2_instances/aws_tag_ec2_instances.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+    "action_title": "Tag AWS Instances",
+    "action_description": "Tag AWS Instances",
+    "action_type": "LEGO_TYPE_AWS",
+    "action_entry_function": "aws_terminate_instance",
```
Comment (shloka-bhalgat-unskript) [inline AWS/legos/aws_tag_ec2_instances/aws_tag_ec2_instances.json:5]: Entry function incorrect

**Review by shloka-bhalgat-unskript — DISMISSED** [review]

File: AWS/legos/aws_tag_ec2_instances/aws_tag_ec2_instances.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    res = ec2.create_tags(Resources=[instance], Tags=[{'Key': tag_key, 'Value': tag_value}])
+
+
+    return(res)
+
+task = Task(Workflow())
+
+task.configure(inputParamsJson='''{
+  
+    "tag_key": "\\"CostCenter1\\"",
```
Comment (amit-chandak-unskript) [inline AWS/legos/aws_tag_ec2_instances/aws_tag_ec2_instances.py:52]: Why do we have this code in the lego @dougsillars ??

File: GCP/legos/gcp_list_vms_access/gcp_list_vms_access.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+    "action_title": "List all GCP VMs and if Publicly Accessible",
+    "action_description": "List all GCP buckets",
```
Comment (amit-chandak-unskript) [inline GCP/legos/gcp_list_vms_access/gcp_list_vms_access.json:3]: description is wrong.

File: GCP/legos/gcp_list_vms_access/gcp_list_vms_access.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    for vm in vms:
+        vm_info = {}
+        vm_info['name'] = vm.name
+        vm_info['publicly_accessible'] = vm.can_ip_forward
+        vm_list.append(vm_info)
+
+    return(vm_list)
+
+
+task = Task(Workflow())
```
Comment (amit-chandak-unskript) [inline GCP/legos/gcp_list_vms_access/gcp_list_vms_access.py:48]: This code should not be there.

**Review by amit-chandak-unskript — CHANGES_REQUESTED** [review]
Take care of comments

File: AWS/legos/aws_tag_ec2_instances/aws_tag_ec2_instances.py (hunk 3 above)
Comment (dougsillars) [reply AWS/legos/aws_tag_ec2_instances/aws_tag_ec2_instances.py:52]: These are essentially default parameters.

Comment (dougsillars) [reply AWS/legos/aws_tag_ec2_instances/aws_tag_ec2_instances.py:52]: Actually the whole task section should be removed.  You can see when these were exported :D

This is resolved

**Comment by dougsillars** [comment]
@amit-chandak-unskript This is ready for re-review.

File: GCP/legos/gcp_list_public_buckets/gcp_list_public_buckets.json [hunk 6]
```diff
@@ -6,6 +6,7 @@
     "action_needs_credential": true,
     "action_supports_poll": true,
     "action_supports_iteration": true,
+    "action_output_type":"ACTION_OUTPUT_TYPE_LIST",
```
Comment (amit-chandak-unskript) [inline GCP/legos/gcp_list_public_buckets/gcp_list_public_buckets.json:9]: As a follow up issue, lets make this action as a check

**Review by amit-chandak-unskript — APPROVED** [review]

**Pull request merged**